src/disc/Albany_DOFManager.hpp

```diff
@@ -0,0 +1,93 @@
+//*****************************************************************//
+//    Albany 3.0:  Copyright 2016 Sandia Corporation               //
+//    This Software is released under the BSD license detailed     //
+//    in the file "license.txt" in the top-level Albany directory  //
+//*****************************************************************//
+
+#ifndef ALBANY_DOF_MANAGER_HPP
+#define ALBANY_DOF_MANAGER_HPP
+
+#include "Albany_GlobalLocalIndexer.hpp"
+#include "Albany_ThyraTypes.hpp"
+#include "Albany_KokkosTypes.hpp"
+
+#include "Panzer_DOFManager.hpp"
+#include "Albany_ConnManager.hpp"
+#include "Teuchos_RCP.hpp"
+
+namespace Albany {
+
+class DOFManager : public panzer::DOFManager {
+public:
+  // Initializes DOF manager
+  DOFManager (const Teuchos::RCP<ConnManager>& conn_mgr,
+              const Teuchos::RCP<const Teuchos_Comm>& comm);
+
+  void build ();
+
+  // Returns (elem_LID,idof)->dof_lid numbering
+  const DualView<const int**>& elem_dof_lids () const;
```

Comment:
Are these instances of the lightweight DualView that you added in Albany_KokkosTypes? Maybe it would be good to add something to differentiate this from the Kokkos DualView to avoid confusion down the road.

Reply:
Yeah, that's it. I guess I could change the name of the type, I just don't have any good idea (all combination I can think of are clunky), so suggestions are welcome. I also thought of using Kokkos's dual view, but in order to sync the two views, one needs to explicitly mark them as modified, and I didn't want to bother. But perhaps, especially considering we don't update these often (ever?), I could really use kokkos dual views...

Reply:
Also, I guess the same could be said for `Albany::ConnManager` and `Albany::DOFManager`, which may be confused with their panzer base classes. I can of course prepend Albany to both of them, but it makes them a bit longer. Perhaps ConnMgr and DOFMgr, though they might be too concise?

Reply:
I feel like at least with the Conn manager and DOF manager, the context is probably enough to know what they are our implementations. But if we start using Kokkos DualViews when removing UVM dependence, things could get messy without prepending `Albany::`. We could cut down on line bloat by just having a single `using dview = Albany::DualView..` like you've used elsewhere. Then at least it is made explicit somewhere in the file.

Reply:
I guess I don't use the type name `DualView` that often, so probably I can prepend Albany:: in those few places where I do. I am already doing it in some places, so might as well do it everywhere.